### PR TITLE
Add 'other roles' to sig-governance.md 

### PR DESCRIPTION
**Pull request by parispittman**
discussion from many issues including #5722 #4289 that will help with elevating folks in SIG roles from a visibility standpoint and other sigs adopting said roles

I started cleaning up some other bullets with formatting and I'll continue after discussion on this.

### CODE REVIEW

File: committee-steering/governance/sig-governance.md [hunk 1]
```diff
@@ -169,6 +170,30 @@ curation from other SIG participants
   - Defined in `SECURITY_CONTACTS` files, this is only relevant to the root file
    in the repository. Template [SECURITY_CONTACTS]
 
+### Other Roles
+This governance document outlines the required roles for SIGs: Chair and Tech
+Lead; however, SIGs are allowed to operate how they see fit outside of minimum 
+governance requirements, including defining more roles to sustain the group. If 
+a SIG needs to change the Chair and Tech Lead position to include or remove
+duties, this needs to be aprpoved by the Steering Committee. Newly created roles
```
Comment (dims) [inline committee-steering/governance/sig-governance.md:178]: `aprpoved` typo. 

LGTM!

File: committee-steering/governance/sig-governance.md [hunk 2]
```diff
@@ -169,6 +170,30 @@ curation from other SIG participants
...
+- [API Reviewer and Moderator]   
+- [Production Readiness Reviewer]  
+- [Events Lead]  
+- [PR Wrangler] 
+- [Editor]
+
+Other roles...
+- *MUST* be tracked on the SIGs README with a link to the role definition
+- *MAY* Fill in for another named role on a temporary basis 
+- *MUST* have the Steering Committees approval to proceed with roles that assume
```
Comment (dims) [inline committee-steering/governance/sig-governance.md:194]: move the `MUST`'s to the top? LGTM

File: committee-steering/governance/sig-governance.md [hunk 3]
```diff
@@ -7,7 +7,8 @@ In order to standardize Special Interest Group efforts, create maximum
 transparency, and route contributors to the appropriate SIG, SIGs should follow
 these guidelines:
 
-- Create a charter and have it approved according to the [SIG charter process]
+- Have an approved Charter [SIG charter process]
+- Have at least one (1) Chair and one (1) defined Tech Lead
```
Comment (mrbobbytables) [inline committee-steering/governance/sig-governance.md:11]: The TL requirement is a big change, while I am generally in favor it in addition to adding more roles we might want to do that change separately.

File: committee-steering/governance/sig-governance.md (hunk 2 above)
Comment (mrbobbytables) [reply committee-steering/governance/sig-governance.md:194]: +1 with dims suggestion to move the musts to the top 👍

**Review by nikhita — COMMENTED** [review]
From the meeting notes, we had one decided upon one more criteria - sending notice to k-dev

```
Recognizing roles defined in SIGs and recognizing leaders in SIGs
Decision: New roles should not be added to sigs.yaml
Decision: SIGs should be able to introduce new roles
Decision: New roles should be sent to sig/k-dev with a 1 week lazy consensus period
Decision: If a role includes delegation of responsibilities from a sig-governance role, that delegation should be documented in your charter
```

**Review by nikhita — APPROVED** [review]
/lgtm

**Comment by parispittman** [comment]
5s a party, merging.

/hold cancel

**Pull request merged**